django_querysets/my_querysets/my_queries/models.py

```diff
@@ -26,3 +28,86 @@ def count_topics(self):
 
     def filter_by_category(self, name):
         return self.filter(category__name=name)
+
+
+class StyleQuerySet(models.QuerySet):
+    def delete_style(self, style_id):
+        return self.filter(id=style_id).update(deleted_at=timezone.now())
+
+
+class AuthorQuerySet(models.QuerySet):
+    def delete_author(self, author_id):
+        return self.filter(id=author_id).update(deleted_at=timezone.now())
+
+
+class BookQuerySet(models.QuerySet):
+    def delete_book(self, book_id):
+        return self.filter(id=book_id).update(deleted_at=timezone.now())
+
+    def sort_authors(self):
+        return self.prefetch_related('authors').order_by('authors__name').values_list('authors__name', flat=True) \
+            .distinct()
+
+    def get_authors_by_country(self, country):
+        return self.prefetch_related('authors').filter(authors__country=country,
+                                                       authors__age__gt=30,
+                                                       authors__deleted_at=None) \
+            .values_list('authors__name', flat=True).distinct()
+
+    def get_books(self, author_name):
+        return self.prefetch_related('authors').filter(authors__name=author_name, authors__deleted_at=None)
+
+    def get_books_by_language(self, author_name, language):
+        return self.prefetch_related('authors').filter(language=language,
+                                                       authors__name=author_name,
+                                                       authors__deleted_at=None
+                                                       )
+
+    def count_books_by_author(self, author_name):
+        return self.prefetch_related('authors').filter(authors__name=author_name).count()
+
+    def count_books_by_author_and_date(self, author_name, books_date):
+        return self.prefetch_related('authors').filter(authors__name=author_name, year__gt=books_date).count()
+
+    def count_pages(self, author_name):
+        return self.prefetch_related('authors').filter(authors__name=author_name).aggregate(Sum('pages_number'))
+
+    def count_pages_by_style(self, author_name, style):
+        return self.prefetch_related('authors').filter(authors__name=author_name,
+                                                       style__name=style
+                                                       ).aggregate(Sum('pages_number'))
+
+    def sort_by_name(self):
```

Comment:
Instead of having two functions that do the same thing add an argument desc=True & False and return query based on this argument